builder/oracle/classic/step_snapshot.go

```diff
@@ -44,6 +47,9 @@ func (s *stepSnapshot) Run(_ context.Context, state multistep.StateBag) multiste
 func (s *stepSnapshot) Cleanup(state multistep.StateBag) {
 	// Delete the snapshot
 	ui := state.Get("ui").(packer.Ui)
+	if !state.Get("cleanup_snap").(bool) {
```

Comment:
if this is just used inside this step, I would make it a member of the `stepSnapshot` struct instead of using the statebag